### ISSUE TITLE
Update version in source

### ISSUE THREAD

**Issue by wren** [jrnl-org/jrnl#1203]
### Use Case/Motivation

We should update the version number of jrnl in the source after every merge into the `develop` branch. Right now, we only do this update on releases and prereleases. The problem with this, though, is that anyone running jrnl from source (like those installing from this Arch package: https://aur.archlinux.org/packages/jrnl-git/) will report their version as the most recent release, even though that's not really what they are running. This can cause some problems with a user is reporting an issue, since it'll take extra work to determine if they are running from source, and what exact version they are on.

**Comment by samuelgregorovic** [comment]
@wren do you mean the `jrnl/__version__.py` ? i could do it via github action. concerning the running from source problem, do you have any ideas how we could manage this?

**Comment by wren** [comment]
Yup! That's the file. I'm not sure what this should look like, though. I was thinking that the version in source might look something like `v2.8-source-12` maybe? The number at the end would be how many commits since that version.

Another idea would be to use a nightly build sort of versioning the source.

Or maybe we introduce a second variable alongside `__version__`. Something to store where jrnl was installed from (like `pip` or `source` or whatever). Although, this would require each package maintainer to edit a file to put their package manager name into it, so I'm not sure if this would fly or not.

Or we could use the git commit hash, maybe?

As you might be able to tell, I haven't quite figured out exactly what this should look like. Any suggestions?

**Comment by samuelgregorovic** [comment]
@wren so, for example, if we would use commit prefix in version, the github action would be incrementing the patch portion of the version number, which would also contain the `source` prefix? 

if pull request is merged into the `develop` branch, and current content of `__version__.py` file in `develop` is e.x. `__version__ = "v2.8.1"`, then after the merge the content would change to `__version__ = "v2.8-source-2"`? 

i just want to make sure i understand how do you want it to work

**Comment by wren** [comment]
@samuelgregorovic I'm not really sure what we want to do for this right now. We should gather ideas in discussion #1305  before starting on this.

**Comment by samuelgregorovic** [comment]
okay. i agree

**Comment by wren** [comment]
Closing this because it's seems like way more trouble than it's worth. If anyone comes up with an elegant solution, please let us know, and I'd be happy to reopen.
